onnxruntime/core/framework/session_state_utils.cc

```diff
@@ -183,20 +249,32 @@ common::Status SaveInitializedTensors(
   }
 
   // tensors requiring a specific allocation order are traced first, to ensure they are allocated in order
+  // NB: vector with init allocation order may contain a subset of all tensors (or none at all)
   auto initialized_tensors_to_allocate = id_to_initialized_tensor;
   for (int ort_value_index : initializer_allocation_order) {
     const auto entry = initialized_tensors_to_allocate.find(ort_value_index);
-    // can not trace string tensor
-    ORT_ENFORCE(entry != initialized_tensors_to_allocate.end() && entry->second->data_type() != ONNX_NAMESPACE::TensorProto_DataType_STRING);
-    ORT_RETURN_IF_ERROR(planner.Trace(entry->first, entry->second));
+    auto location = exec_plan.GetLocation(ort_value_index).name;
+    if (utils::HasExternalData(*entry->second) && (strcmp(location, CPU) == 0)) {
+      // exernal data will be memory mapped, no need to plan for its allocation
```

Comment:
Nit: exernal -> external . And do you want to call out that why we are limiting it to just CPU initializers ?